apps/communications/contacts/views/form/js/form_controller.js

```diff
@@ -0,0 +1,201 @@
+/* global LazyLoader, MergeHelper, ContactsService, utils,
+   Matcher, MatchService */
+(function(exports) {
+  'use strict';
+
+  var _;
+  var _activity;
+  const CONTACTS_APP_ORIGIN = location.origin;
+
+  function close() {
+    if (_activity) {
+      _activity.postResult({});
+    } else {
+      window.history.back();
+    }
+  }
+
+  function hasName(contact) {
+    return (Array.isArray(contact.givenName) && contact.givenName[0] &&
+              contact.givenName[0].trim()) ||
+            (Array.isArray(contact.familyName) && contact.familyName[0] &&
+              contact.familyName[0].trim());
+  }
+
+
+  function doMerge(contact, list, cb) {
+    var callbacks = {
+      success: cb,
+      error: function(e) {
+        console.error('Failed merging duplicate contacts: ', e.name);
+        cb();
+      }
+    };
+
+    LazyLoader.load('/contacts/js/utilities/merge_helper.js', function() {
+      MergeHelper.merge(contact, list).then(callbacks.success, callbacks.error);
+    });
+  }
+
+  function getCompleteName(contact) {
+    var givenName = Array.isArray(contact.givenName) ?
+                    contact.givenName[0] : '';
+
+    var familyName = Array.isArray(contact.familyName) ?
+                    contact.familyName[0] : '';
+
+    var completeName = givenName && familyName ?
+                       givenName + ' ' + familyName :
+                       givenName || familyName;
+
+    return completeName;
+  }
+  // Fills the contact data to display if no givenName and familyName
+  function getDisplayName(contact) {
+    if (hasName(contact)) {
+      return { givenName: contact.givenName, familyName: contact.familyName };
+    }
+
+    var givenName = [];
+    if (Array.isArray(contact.name) && contact.name.length > 0) {
+      givenName.push(contact.name[0]);
+    } else if (contact.org && contact.org.length > 0) {
+      givenName.push(contact.org[0]);
+    } else if (contact.tel && contact.tel.length > 0) {
+      givenName.push(contact.tel[0].value);
+    } else if (contact.email && contact.email.length > 0) {
+      givenName.push(contact.email[0].value);
+    } else {
+      givenName.push(_('noName'));
```

Comment:
As you know we are keeping the code as it's is by now, but I've created a bug to remove all deprecated methods in #new[1]. I've added the bug as a comment and the dependency in bugzilla.

[1] https://bugzilla.mozilla.org/show_bug.cgi?id=1181586